operator/internal/controller/auto/update_controller.go

```diff
@@ -168,6 +192,36 @@ func (r *UpdateReconciler) Reconcile(ctx context.Context, req ctrl.Request) (ctr
 	}
 }
 
+func isWorkspaceReady(ws *autov1alpha1.Workspace) bool {
+	if ws == nil || ws.Generation != ws.Status.ObservedGeneration {
+		return false
+	}
+	return meta.IsStatusConditionTrue(ws.Status.Conditions, autov1alpha1.WorkspaceReady)
+}
+
+type workspaceReadyPredicate struct {
+	predicate.Funcs
+}
```

Comment:
No need to embed this.
```suggestion
type workspaceReadyPredicate struct {}
```

Reply:
Ah true, I was only extending the Update at first, but now am implementing all the interface methods.